central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -312,11 +414,56 @@ func (m *managerImpl) handleReadyScanConfig() {
 			return
 		default:
 			if result := m.scanConfigReadyQueue.PullBlocking(m.stopper.LowLevel().GetStopRequestSignal()); result != nil {
-				log.Infof("Scan Config %s done with %d scans", result.ScanConfig.GetScanConfigName(), len(result.ScanResults))
+				log.Debugf("Scan Config %s done with %d scans and %d reports", result.ScanConfig.GetScanConfigName(), len(result.ScanResults), len(result.ReportSnapshot))
 				concurrency.WithLock(&m.watchingScanConfigsLock, func() {
 					delete(m.watchingScanConfigs, result.WatcherID)
 				})
+				go m.generateReportsFromWatcherResults(result)
```

Comment:
I see in other places that we start `m.generateReport(generateReportReq)` in a goroutine, however here we start one goroutine for multiple reports. I guess it is safe as `result` should not be changed anymore after pulling it from the queue, but maybe we can also run `go m.generateReport(generateReportReq)` for some performance?

Reply:
I've refactored a few things in that logic. Let me know what you think of the changes.